Gems/ROS2/Code/Tests/GNSSTest.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzTest/AzTest.h>
+
+#include "GNSS/GNSSFormatConversions.h"
+
+namespace UnitTest
+{
+
+    class GNSSTest : public AllocatorsTestFixture
+    {
+    };
+
+    TEST_F(GNSSTest, WGS84ToECEF)
+    {
+        const AZStd::vector<AZStd::pair<AZ::Vector3, AZ::Vector3>> input_gold_set = {
+            { { 10.0f, 20.0f, 300.0f }, { 5903307.167667380f, 2148628.092761247f, 1100300.642188661f } },
+            { { -70.0f, 170.0f, 500.0f }, { -2154856.524084172f, 379959.3447517005f, -5971509.853428957f } },
+            { { 50.0f, -120.0f, -100.0f }, { -2053899.906222906f, -3557458.991239029f, 4862712.433262121f } },
+        };
+        for (const auto& [input, goldResult] : input_gold_set)
+        {
+            AZ::Vector3 result = ROS2::GNSS::WGS84ToECEF(input);
+            EXPECT_NEAR(result.GetX(), goldResult.GetX(), 1.0f);
+            EXPECT_NEAR(result.GetY(), goldResult.GetY(), 1.0f);
+            EXPECT_NEAR(result.GetZ(), goldResult.GetZ(), 1.0f);
+        }
+    }
+
+    TEST_F(GNSSTest, ECEFToENU)
+    {
+        const AZStd::vector<AZStd::tuple<AZ::Vector3, AZ::Vector3, AZ::Vector3>> input_gold_set = {
```

Comment:
The variable name should use lowerCamelCase.
```suggestion
        const AZStd::vector<AZStd::tuple<AZ::Vector3, AZ::Vector3, AZ::Vector3>> inputGoldSet = {
```